lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FileWatchService.java

```diff
@@ -0,0 +1,149 @@
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.tools.langserver.lemminx.services;
+
+import io.openliberty.tools.langserver.lemminx.util.LibertyConstants;
+import io.openliberty.tools.langserver.lemminx.util.LibertyUtils;
+import org.apache.commons.io.IOCase;
+import org.apache.commons.io.filefilter.FileFilterUtils;
+import org.apache.commons.io.filefilter.IOFileFilter;
+import org.apache.commons.io.filefilter.NameFileFilter;
+import org.apache.commons.io.filefilter.SuffixFileFilter;
+import org.apache.commons.io.monitor.FileAlterationListenerAdaptor;
+import org.apache.commons.io.monitor.FileAlterationMonitor;
+import org.apache.commons.io.monitor.FileAlterationObserver;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public class FileWatchService {
+
+    private final Set<FileAlterationObserver> fileObservers = new HashSet<>();
+    private final Set<FileAlterationMonitor> monitors = new HashSet<>();
+
+    private static final FileWatchService instance = new FileWatchService();
+
+    public static FileWatchService getInstance() {
+        return instance;
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(FileWatchService.class.getName());
+
+    private FileWatchService() {
+    }
+
+    /**
+     * add observer for file changes
+     *
+     * @param workspace  workspace
+     * @param watchLocations parent locations
+     */
+    public void addFileAlterationObserver(LibertyWorkspace workspace, List<String> watchLocations)
+            throws Exception {
+        for (String location:watchLocations) {
+            FileAlterationObserver observer = getFileAlterationObserver(location, workspace);
+            observer.initialize();
+            fileObservers.add(observer);
+            FileAlterationMonitor monitor = new FileAlterationMonitor();
+            monitor.addObserver(observer);
+            monitor.start();
+            monitors.add(monitor);
+        }
+    }
+
+    private FileAlterationObserver getFileAlterationObserver(final String parentPath, LibertyWorkspace workspace) {
+        //ignore files/directories with below suffixes and names
+        IOFileFilter notFileFilter = FileFilterUtils.notFileFilter(
+                new SuffixFileFilter(Arrays.asList(".class", ".lst", ".txt", ".log", ".manager", ".libertyls",
+                        ".sLock", ".jar", ".war", ".ear",".mf"),
+                        IOCase.INSENSITIVE)
+                        .or(new NameFileFilter(Arrays.asList("workarea", "plugin-cfg.xml", "libs", "tmp", "classes","bin",
```

Comment:
Is `plugin-cfg.xml` really a directory? Seems like you changed this logic...are these just files now? Before you added files in one part and directories in another. Is there no distinction?

Reply:
NameFileFilter accepts both files and directories
And it looks for base names from the path, but only considers the path after our watch location. In below example, we watch target folder and I have set path for my workspace as  "/Users/arunkumarvn/Downloads/libs/app-name/target/liberty/wlp"

when an item present in our namefilter is found, ie generated-sources, accept() returns true
![image](https://github.com/user-attachments/assets/17ce9e29-74b6-491f-a7cb-94de4a3729c8)
![image](https://github.com/user-attachments/assets/eefe6a63-311e-4b4c-a6cf-80548c933133)
![image](https://github.com/user-attachments/assets/df18d687-680a-4c1b-a1c7-6e9d3ff0a2b0)


whereas if the current file is not part of any dir/file specified in namefilter, accept() returns false
![image](https://github.com/user-attachments/assets/ab9ba61d-b9cf-471e-ae38-a66ab5ef9a1b)

![image](https://github.com/user-attachments/assets/2c31322a-6452-42bd-9d81-f6640d9c571c)